spec.bs

```diff
@@ -509,6 +510,50 @@ are:
 
 </div>
 
+<h3 id="clearoriginjoinedAdInterestGroups">clearOriginJoinedAdInterestGroups()</h3>
+{{Window/navigator}}.{{Navigator/clearOriginJoinedAdInterestGroups()}} removes a user from
+[=interest groups=] whose [=interest group/joining origin=] is the current top level page's
+origin.
+
+
+<xmp class="idl">
+[SecureContext]
+partial interface Navigator {
+  Promise<undefined> clearOriginJoinedAdInterestGroups(
+      USVString owner, optional sequence<USVString> interestGroupsToKeep = []);
+};
+</xmp>
+
+<div algorithm>
+
+The <dfn for=Navigator method>clearOriginJoinedAdInterestGroups(|owner|, |interestGroupsToKeep|)</dfn>
+method steps are:
+
+1. Let |frameOrigin| be [=this=]'s [=relevant settings object=]'s
+  [=environment settings object/origin=].
+1. [=Assert=] that |frameOrigin| is not an [=opaque origin=] and its [=origin/scheme=] is "`https`".
+1. Let |p| be [=a new promise=].
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the
+  "[=join-ad-interest-group=]" [=policy-controlled feature=], then [=exception/throw=] a
+  "{{NotAllowedError}}" {{DOMException}}.
+
+  Note: Both joining and leaving interest groups use the "join-ad-interest-group" feature.
+1. Let |ownerOrigin| be the result of [=parsing an https origin=] with |owner|.
+1. If |ownerOrigin| is failure, [=exception/throw=] a {{TypeError}}.

```

Comment:
Do you know of any cases where it's possible for the current origin of a valid HTTPs document to fail parsing? I'm just trying to see if this is possible to hit.

Reply:
Note that this is not the origin of the document, but a parameter to clearOriginJoinedAdInterestGroups(), so it can fail to parse.

Reply:
Ah right. Thanks.